metrics/internal/cache/cephfs_pv.go

```diff
@@ -0,0 +1,291 @@
+package cache
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/red-hat-storage/ocs-operator/metrics/v4/internal/options"
+	rookclient "github.com/rook/rook/pkg/client/clientset/versioned"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+)
+
+// Cache metadata for all PV with annotation - pv.kubernetes.io/provisioned-by: *.cephfs.csi.ceph.com
+
+var _ cache.Store = &CephFSPVStore{}
+
+type CephFSPVAttributes struct {
+	PersistentVolumeName string
+}
+
+type CephFSPVStore struct {
+	Mutex sync.RWMutex
+	Store map[types.UID]CephFSPVAttributes
+	// CephFSSubvolumeCountMap is a map of "subvolumegroup" to subvolume count
+	CephFSSubvolumeCountMap   map[string]int
+	monitorConfig             cephMonitorConfig
+	kubeClient                clientset.Interface
+	rookClient                rookclient.Interface
+	cephClusterNamespace      string
+	cephAuthNamespace         string
+	runCephfsSubvolumeCountFn func(config cephMonitorConfig, rookClient rookclient.Interface, cephClusterNamespace string) (map[string]int, error)
+}
+
+func NewCephFSPVStore(opts *options.Options) *CephFSPVStore {
+	return &CephFSPVStore{
+		Store:                     map[types.UID]CephFSPVAttributes{},
+		CephFSSubvolumeCountMap:   make(map[string]int),
+		kubeClient:                clientset.NewForConfigOrDie(opts.Kubeconfig),
+		rookClient:                rookclient.NewForConfigOrDie(opts.Kubeconfig),
+		cephClusterNamespace:      opts.AllowedNamespaces[0],
+		cephAuthNamespace:         opts.CephAuthNamespace,
+		monitorConfig:             cephMonitorConfig{},
```

Comment:
i think cephMonitorConfig is never initialised, we need to run `initCeph` for it to be initialised